plugins/modules/aws_msk_config.py

```diff
@@ -0,0 +1,304 @@
+        all_configs = get_configurations_with_backoff(client)["Configurations"]
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="failed to obtain kafka configurations")
+
+    active_configs = list(
+        item
+        for item in all_configs
+        if item["Name"] == name and item["State"] == "ACTIVE"
+    )
+
+    if active_configs:
+        if len(active_configs) == 1:
+            return active_configs[0]
+        else:
+            module.fail_json_aws(
+                msg="found more than one active config with name '{0}'".format(name)
+            )
+
+    return None
+
+
+def create_config(client, module):
+    """create new or update existing configuration"""
+
+    config = find_active_config(client, module)
+
+    # create new configuration
+    if not config:
+
+        if module.check_mode:
+            return True, {}
+
+        try:
+            response = client.create_configuration(
+                Name=module.params.get("name"),
+                Description=module.params.get("description"),
+                KafkaVersions=module.params.get("kafka_versions"),
+                ServerProperties=dict_to_prop(module.params.get("config")).encode(),
+            )
+        except (
+            botocore.exceptions.BotoCoreError,
+            botocore.exceptions.ClientError,
+        ) as e:
+            module.fail_json_aws(e, "failed to create kafka configuration")
+
+    # update existing configuration (creates new revision)
+    else:
+        # it's required because 'config' doesn't contain 'ServerProperties'
+        response = client.describe_configuration_revision(
+            Arn=config["Arn"], Revision=config["LatestRevision"]["Revision"]
+        )
+
+        # compare configurations (description and properties) and update if required
+        # prop_module = {str(k): str(v) for k, v in module.params.get("config").items()}
+        prop_module = {}
+        for k, v in module.params.get("config").items():
+            prop_module[str(k)] = str(v)
+        if prop_to_dict(response.get("ServerProperties", "")) == prop_module:
+            if response.get("Description", "") == module.params.get("description"):
+                return False, response
+
+        if module.check_mode:
+            return True, {}
+
+        try:
+            response = client.update_configuration(
+                Arn=config["Arn"],
+                Description=module.params.get("description"),
+                ServerProperties=dict_to_prop(module.params.get("config")).encode(),
+            )
+        except (
+            botocore.exceptions.BotoCoreError,
+            botocore.exceptions.ClientError,
+        ) as e:
+            module.fail_json_aws(e, "failed to update kafka configuration")
+
+    arn = response["Arn"]
+    revision = response["LatestRevision"]["Revision"]
+
+    result = client.describe_configuration_revision(Arn=arn, Revision=revision)
```

Comment:
Please wrap in try/except and add retries

Reply:
fixed